src/libraries/Microsoft.PowerFx.Core/Utils/MaxCallDepthException.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.PowerFx.Core.Utils
+{
+    [Serializable]
+    internal class MaxCallDepthException : Exception
```

Comment:
>MaxCallDepthException [](http://example.com/codeflow?start=19&length=21)

What did we settle on for SDK model here ... 

Does Engine.Eval()  throw this exception (and hosts catch it)? Or does it get converter to some runtime error at top-level?

Reply:
We decided that it would get converted to a runtime error at the top level, top of the EvalAsync function. Reimplemented now.